pkg/virt-handler/vm.go

```diff
@@ -874,18 +874,16 @@ func (d *VirtualMachineController) updateVMIStatus(origVMI *v1.VirtualMachineIns
 			// Set Pod Interface
 			interfaces := make([]v1.VirtualMachineInstanceNetworkInterface, 0)
 			for _, network := range vmi.Spec.Networks {
-				if network.NetworkSource.Pod != nil {
-					podIface, err := d.getPodInterfacefromFileCache(vmi, network.Name)
-					if err != nil {
-						return err
-					}
-					ifc := v1.VirtualMachineInstanceNetworkInterface{
-						Name: network.Name,
-						IP:   podIface.PodIP,
-						IPs:  podIface.PodIPs,
-					}
-					interfaces = append(interfaces, ifc)
+				podIface, err := d.getPodInterfacefromFileCache(vmi, network.Name)
```

Comment:
Please add a unit test.

Reply:
I'd take this a step further - we need more than one. To be on the safe side of things, I'd like to a see: 
  - a simple unit test where it shows you picking up the right IP on a secondary interface.
  - a unit test where we handle the basic scenario + have guest agent. I am expecting the guest agent to just override whatever is cached - could this be an issue ?